homeassistant/components/duwi/__init__.py

```diff
@@ -0,0 +1,380 @@
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import __version__
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers import config_validation as cv, device_registry as dr
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    _LOGGER,
+    ACCESS_TOKEN,
+    ADDRESS,
+    APP_KEY,
+    APP_SECRET,
+    CLIENT_MODEL,
+    CLIENT_VERSION,
+    DOMAIN,
+    DUWI_DISCOVERY_NEW,
+    DUWI_HA_SIGNAL_UPDATE_ENTITY,
+    DUWI_SCENE_UPDATE,
+    HOUSE_KEY,
+    HOUSE_NAME,
+    HOUSE_NO,
+    PASSWORD,
+    PHONE,
+    REFRESH_TOKEN,
+    SUPPORTED_PLATFORMS,
+    WS_ADDRESS,
+)
+
+type DuwiConfigEntry = ConfigEntry[HomeAssistantDuwiData]
+
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(domain=DOMAIN)
+
+
+class HomeAssistantDuwiData(NamedTuple):
+    """Duwi data stored in the Home Assistant data object."""
+
+    manager: Manager
+    listener: SharingDeviceListener
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Duwi Smart Devices integration."""
+
+    # Check for existing config entries for this integration
+    hass.data.setdefault(DOMAIN, {})
+    if not hass.config_entries.async_entries(DOMAIN):
+        # No entries found, initiate the configuration flow
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data={}
+            )
+        )
+
+    # Setup was successful
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: DuwiConfigEntry) -> bool:
+    """Set up a config entry for the Duwi Smart Devices integration."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    token_listener = TokenListener(hass, entry)
+    manager: Manager = Manager(
+        _id=entry.entry_id,
+        customer_api=CustomerApi(
+            address=entry.data[ADDRESS],
+            ws_address=entry.data[WS_ADDRESS],
+            app_key=entry.data[APP_KEY],
+            app_secret=entry.data[APP_SECRET],
+            house_no=entry.data[HOUSE_NO],
+            house_name=entry.data[HOUSE_NAME],
+            access_token=entry.data[ACCESS_TOKEN],
+            refresh_token=entry.data[REFRESH_TOKEN],
+            client_version=CLIENT_VERSION,
+            client_model=CLIENT_MODEL,
+            app_version=__version__,
+        ),
+        house_key=entry.data.get(HOUSE_KEY),
+        token_listener=token_listener,
+    )
+
+    login_status = await manager.init_manager(
+        entry.data["phone"], entry.data["password"]
+    )
+
+    if not login_status:
+        raise ConfigEntryAuthFailed(
+            "User authentication failed, please try reloading the integration or adding again."
+        )
```

Comment:
why would reload help?

Reply:
Thank you for the reminder. I have already modified this part of the code